connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -379,6 +380,20 @@ private void logDeprecatedProperty(String propName, String propValue, String def
         }
     }
 
+    private void logPluginPathConfigProviderWarning(Map<String, String> rawOriginals) {
+        String rawPluginPath = rawOriginals.get(PLUGIN_PATH_CONFIG);
+        String transformedPluginPath = originalsStrings().get(PLUGIN_PATH_CONFIG);
+        if (!Objects.equals(rawPluginPath, transformedPluginPath)) {
+            log.error(
```

Comment:
```suggestion
            log.warn(
```

Reply:
Yeah, I was torn here... on one hand, nothing's necessarily broken, no exceptions are thrown, and the worker won't fail to start up, so `WARN` does seem warranted. On the other hand, it's an unfortunate reality that the `WARN` level is heavily polluted by existing config warnings emitted when the framework creates Kafka clients and their unused configs get logged at that level, and it also seems highly unlikely that the worker would behave as expected in this case.

I think if this were noticed from the first time it became an issue, it might even be reasonable to throw an exception and fail the worker on startup, but since we can't do that now without breaking backwards compatibility and opening that whole can of worms, `ERROR` level logging seemed like a decent substitute.

That's my thinking, if you're still convinced that `WARN` is better I'm happy to change it.

Reply:
I think what you just described is precisely what I would consider something worthy of logging at `WARN` level. i.e. something went wrong and it is very likely that the application will behave in unexpected ways. `ERROR` level to me should be reserved for catastrophic failures that the application can not recover from. Something that would prevent the worker to start up, in this case. But, at the end of the day, everyone has their own interpretations and there is no right or wrong answer here.

Now, `WARN` level being polluted by unused configs - that is something that should be considered a bug that we should fix separately, IMO.

Reply:
Fair enough, will update accordingly :)